### PR TITLE
Allow mod swapping for mod tap (MT)

### PR DESCRIPTION
**Pull request by jackhumbert**
Addresses #1192

### CODE REVIEW

**Comment by phongnh** [comment]
Hi @jackhumbert, Thanks for your fix. However, the problem has not been fixed yet.

### **On both Mac and Windows:**
When AG_SWAP is enabled:
- KC_LALT worked as KC_LGUI
- ALT_T worked as KC_LGUI
- KC_LGUI worked as KC_LALT
- **GUI_T did not work as KC_LALT**

File: quantum/keycode_config.c [hunk 1]
```diff
@@ -88,3 +88,32 @@ uint16_t keycode_config(uint16_t keycode) {
...
     }
 }
+
+uint8_t mod_config(uint8_t mod) {
+    if (keymap_config.swap_lalt_lgui) {
+        if (mod & MOD_LGUI) {
+            mod &= ~MOD_LGUI;
+            mod |= MOD_LALT;
+        }
+        if (mod & MOD_LALT) {
```
Comment (fredizzimo) [inline quantum/keycode_config.c]: I just took a very quick look, and I happened to notice this.

Changing this to `else if` should fix the issue. The block below also has the same issue so it should be changed too.

Comment (jackhumbert) [inline quantum/keycode_config.c]: Doy, thanks :)

**Comment by jackhumbert** [comment]
It looks like the changes suggested by @fredizzimo should fix that! If you get the chance, @phongnh, would you mind checking things again?

**Comment by phongnh** [comment]
I'm sorry I cannot test it now. I will test it asap I come back my home. 
Thanks @jackhumbert  and @fredizzimo

**Comment by phongnh** [comment]
Tried to test again on both Mac and Windows. The problem is still there.

### Mac:
- KC_LALT and KC_LGUI are swapped correctly
- ALT_T and GUI_T are not swapped, ALT_T stills behave as LALT and GUI_T stills behave as LGUI

### Windows:
- KC_LALT and KC_LGUI are swapped correctly
- KC_LALT + Tab does not work
- ALT_T and GUI_T are not swapped, ALT_T stills behave as LALT and GUI_T stills behave as LGUI

Thanks.

**Comment by fredizzimo** [comment]
Looking at this code 
```C
mod = mod_config((keycode >> 0x8) & 0x1F);
action.code = ACTION_MODS_TAP_KEY(mod, keycode & 0xFF);
```

When we have
```C
#define ACTION_MODS_TAP_KEY(mods, key)  ACTION(ACT_MODS_TAP, ((mods)&0x1f)<<8 | (key))
```

The `|` of `ACTION_MODS_TAP_KEY`, wont completely overwrite the old mods, so we need this

```C
mod = mod_config((keycode >> 0x8) & 0x1F);
action.code = ACTION_MODS_TAP_KEY(mod, keycode & (~(0x1F << 8)));
```
To properly clear out the old modifiers

**Comment by jackhumbert** [comment]
@phongnh sorry for the delay here - I think I found the bug! It seems to be working for me here - would you mind testing things when you have the chance?

**Comment by phongnh** [comment]
Thanks @jackhumbert. I will check it as soon as I'm back my home.

**Comment by phongnh** [comment]
Hi @jackhumbert 

The issue is still standing there. It does not work in mac. GUI_T and ALT_T are not working. Maybe you just use `keycode_config` function to change key before putting it to `MOD_BIT`. I tried to work around this problem as above, please check https://github.com/phongnh/whitefox-settings/blob/master/keymap.c#L116-L132. 

Thank you for your support.

**Comment by jackhumbert** [comment]
Sorry, I pushed that last commit without testing - the one prior worked (almost)!

The latest commits should concretely fix this, though. Sorry for all the hassle here!

**Comment by phongnh** [comment]
@jackhumbert, I tried to recompile with latest commit. It works well in macOS but not Windows. On Windows, KC_LGUI and GUI_T are swapped correctly. KC_LATL and ALT_T also are swapped correctly. However, KC_LATL + Tab and ALT_T + Tab do not work anymore in both AG_NORM and AG_SWAP.

Thank you.

**Comment by jackhumbert** [comment]
Your comment before the edit scared me a little :) thanks for your patience and willingness to test!

**Comment by T-800** [comment]
Thank you :)

**Comment by phongnh** [comment]
Thank you :)

**Pull request merged**